dashboard/signals.py

```diff
@@ -0,0 +1,19 @@
+"""
+Signals for user profiles
+"""
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from dashboard.models import CachedEnrollment, ProgramEnrollment
+
+
+@receiver(post_save, sender=CachedEnrollment, dispatch_uid="update_programenrollment")
+def create_programenrollment(sender, instance, created, **kwargs):  # pylint: disable=unused-argument
+    """
+    Signal handler to create Program enrollment when the CachedEnrollment table is updated
+    """
+    if instance.data is not None:
+        ProgramEnrollment.objects.get_or_create(
+            user=instance.user,
+            program=instance.course_run.course.program
+        )
```

Comment:
if a student enrolls in a single course in a program then withdraws, will we want to keep the ProgramEnrollment record?